Merge Sekiro world into Main branch (#68)
nex3/Archipelago: worlds/sekiro/docs/locations_en.md

  * [PP: Poison Pool](#poison-pool)
  * [HF: Hidden Forest](#hidden-forest)
  * [MV: Mibu Village](#mibu-village)
  * [AC/I: Ashina Castle (Interior Ministry)](#ashina-castle-interior-ministry)
  * [HE2: Hirata Estate (Father's Bell Charm)](#hirata-estate-fathers-bell-charm)
  * [FP1: Fountainhead Palace (before underwater progression)](#fountainhead-palace-before-underwater-progression)
  * [FP2: Fountainhead Palace (underwater progression)](#fountainhead-palace-underwater-progression)
  * [AC/C: Ashina Castle (Central Forces)](#ashina-castle-central-forces)
  * [AO/C: Ashina Outskirts (Central Forces)](#ashina-outskirts-central-forces)
* [Detailed Location Descriptions](#detailed-location-descriptions)

## Location Groups

The Sekiro: Shadows Die Twice randomizer supports a number of location group names, which
can be used in YAML options like `exclude_locations` to refer to many locations
at once:

* **Prominent:** A small number of locations that are in very obvious locations.
  Mostly boss drops. Ideal for setting as priority locations.

* **Progression:** Locations that contain items in vanilla which unlock other locations.

> * **Incense:** Locations that contain items required to reach the Divine Realm (Fountainhead Palace) in vanilla.

Reviewer: I don't think this is worth being a separate location group from `Progression`.

Author: Makes sense, I think that was just copied from the Cinders in DS3.

Will change.

* **Boss Rewards:** Boss drops. Bosses are strong enemies that drop memories.

> * **Miniboss Rewards:** Miniboss drops. Minibosses are enemies with unique health bars that do not drop memories.

Reviewer: ```suggestion
* **Miniboss Rewards:** Miniboss drops. Minibosses are enemies with boss bars that do not drop memories.
```

Same below.

* **Hostile:** Drops from regular enemies or NPCs that are hostile to you. 
  This includes initially-friendly NPCs that must be fought as part of their questline.

* **Friendly:** Locations that contain items given by friendly NPCs as part of their quests
  or from non-violent interaction.

* **Esoteric Texts:** Locations that contain an esoteric text item.

> * **Skills:** Locations that contain skills found as item drops, such as Shinobi Medicine or Ninjutsu techniques.

Reviewer: Are there enough of these to be worth their own category? Would it make more sense to lump both Esoteric Texts and Skills into a single category?

Author: It was split because the Esoteric Texts are quite a bit more valuable when using the skill splitter. And since we have no logic yet to actually make the book containing mikiri early, some people may want to make it easier to put those early.

Skills would also be much bigger if we implement the Skills as pickups, so that is why I kept it separate for now.

Reviewer: What do esoteric texts even _do_ when using the skill splitter?

Author: In matts rando and what I'd also be going for is they get removed from the item pool if that option is enabled.

* **Upgrade:** Locations that contain non-unique upgrade materials for prosthetic tools in vanilla.

* **Currency:** Locations that contain coin pouches and treasure carp scales in vanilla.

* **Memories:** Locations that contain memories in vanilla.

* **Unique:** Locations that contain items which can be obtained once in a run, such as keys, notes, 
  reusable items, prosthetics & unique upgrades.

* **Healing:** Locations that contain Gourd Seeds and Prayer Beads in vanilla.

* **Miscellaneous:** Locations that contain generic stackable items such as sugars, mibu balloons, resistance buffs and so on.

* **Hidden:** Locations that are particularly difficult to find, such as in crawl spaces,
  down hidden drops, behind walls and so on. 

* **Offering Box:** Locations that contain items which move into the Offering Box after
  becoming unavailable elsewhere.

## Understanding Location Names

All locations begin with an abbreviation indicating their general region. Most
locations have a set of landmarks that are used in location names to keep them
short.

* **T:** [Tutorial](#tutorial)

> * **DT:** Dilapidated Temple

Reviewer: I think it's worth having a section for this if just to remind people that it's the starting area where the sculptor hangs out.

Author: Will add that.

* **SV:** [Sunken Valley](#sunken-valley)
* **SVP:** [Sunken Valley Passage](#sunken-valley-passage)
* **PP:** [Poison Pool](#poison-pool)
* **HF:** [Hidden Forest](#hidden-forest)
* **MV:** [Mibu Village](#mibu-village)
* **AC/I:** [Ashina Castle (Interior Ministry)](#ashina-castle-interior-ministry)
* **HE2:** [Hirata Estate (Father's Bell Charm)](#hirata-estate-fathers-bell-charm)
* **FP1:** [Fountainhead Palace (before underwater progression)](#fountainhead-palace-before-underwater-progression)
* **FP2:** [Fountainhead Palace (underwater progression)](#fountainhead-palace-underwater-progression)
* **AC/C:** [Ashina Castle (Central Forces)](#ashina-castle-central-forces)
* **AO/C:** [Ashina Outskirts (Central Forces)](#ashina-outskirts-central-forces)
* **AR/C:** Ashina Reservoir (Central Forces)

General notes:

* "Miniboss" refers to enemies with unique health bars that do not drop memories.

* NPC quest items are always in the first location you can get them _without_
  killing the NPC or ending the quest early.

* Currently the logic assumes you send Jinzaemon to Doujun in the Abandoned Dungeon and Kotaro to Anayama.

>   Therefore the locations for Jinza's Jizo Statue and the Taro Persimmon are deemed unreachable and not generated.

Reviewer: Two things:

1. I don't think this is worth noting here because it's not directly relevant to location descriptions.
2. *All* possible rewards for these quests should be marked as missable. We can't assume that a player will do one thing or another; if we don't enforce it (which we can't do for the time being) we have to make the logic entirely pessimistic and treat every location that a player could possibly lock themselves out of as missable. (We can still randomize items there, just not progression or important items.)

Author: 1. It is really just there for the sake of remembering what is currently implemented.
2. The locations of the quests are all marked as missable already. The Taro Persimmon and Jinzas Statue are just not generated so would contain their default item. We can also enable them, I dont mind.

Author: Considering the feedback in the discord I will add these two back in and remove the specific description.

### Tutorial

The area from spawning into the starting well up to losing your arm in the intro boss cutscene.

* **Moon-View Tower:** The pagoda Kuro resides in at the start of the game, where he gives you the Kusabimaru.

### Ashina Outskirts

> This area ends at the Ashina Castle Gate Idol, anything after the gate falls under Ashina Castle. \

Reviewer: I think we should say this ends after the miniboss arena, because technically there are locations between that door and the gate to the castle that I think would be pretty unintuitive as "outskirts".

Also, nit: don't capitalize "idol".

Author: That was a whole discussion if it should be castle or outskirts lol
I dont mind either, Ill change it to Outskirts for now then?

yeah, Idol was also a thing. Will remove the caps.

> "Right" is the side opposite of Ashina Castle in the distance. "Forward" is the path towards progress.

Reviewer: What do you think about defining and using actual cardinal directions rather than purely relative ones? Sekiro's map does correspond decently well to the actual game world and I think to the sun as well, so we can use that as an anchor. Ashina Castle is due east from outskirts, for example.

Author: We could do that, I guess the game doesn't have a compass so it might throw some people off?

Ashina Outskirts I mostly took what was done by whoever maintained the descriptions in the sheet. I just adjusted obvious errors.

Author: On second thought, I think sticking to left and right and just having it be in the direction of area progression is the most clear. Forward I will remove, because that is not used at all..

can rework this again if needed.

* **Gate:** Large torii gates with solid doors.

* **Wide Gate House:** Large gate with a wide second floor, just after the Outskirts Wall - Gate Path Idol.

* **Cliff Courtyard:** The miniboss arena after the Outskirts Wall - Gate Path Idol.

* **Destroyed House:** The two-floor building with a missing roof after the cliff courtyard.

* **Cliffside Gate:** The closed gate flanked by giant cobblestone structures, 
  next to the secluded platforms with the giant snake skin.

* **Cobblestone Platform:** Large rectangular cobblestone platforms beside a giant gate near 
  the Outskirts Wall - Stairway Idol.

* **Ridge Inner Curve:** Much of Ashina Outskirts is on a thin U-shaped mountain ridge, 

>   the inner curve is the "bowl" of the U-shape.

Reviewer: Are there other landmarks we can refer to here?

Author: I will check, this is another one I just took from the maintenance sheet.

Author: Since this only includes two items and one of them is in a hidden cave, I will describe the hidden cave and since the other item is just one grapple away from that entrance, describe that according to the hidden cave.

* **Gun Fort:** Starting from Sunken Valley Idol all the way to the doors unlocked with the Gun Fort Shrine Key.
                Gun Fort itself is the area with the barricades/mines and the cave leading to the Gun Fort Idol.

  * "Approach" is the area starting from Sunken Valley Idol and ending once you climb up to the 
               barricaded area containing mines.
  * "Shrine" is the shrine past the Gun Fort Idol where a miniboss is fought.
  * "Chasm" is the small network of caves accessed through broken floorboards inside the Gun Fort Shrine.

### Sunken Valley Passage

This region starts after proceeding through the Gun Fort Shrine doors with the Gun Fort Shrine Key.

* **Bodhi Statue:** Refers to the many giant statues littering Bodhisattva Valley, right after Riven Cave.
  "Left" and "Right" are from the perspective of the player coming from Riven Cave.
  
* **Swamp Island:** The landmass at the end of the toxic swamp, coming from the Bodhisattva Valley Idol.

* **Shrine Cave:** The cave that contains the underground shrine proceeding past the Toxic Memorial Mob.

* **Underground Shrine:** The underground temple guarded by a Great Serpent.

>   Requires Puppeteer Ninjutsu or Mist Raven's Feathers is needed to pass.

Reviewer 2: Gramatically weird. Either "requires" or "are needed to pass"

Author: Thank you! Definitely missed that. Has been corrected on next commit.

* **AO/C:** [Ashina Outskirts (Central Forces)](#ashina-outskirts-central-forces)
* **AR/C:** Ashina Reservoir (Central Forces)

General notes:

* "Miniboss" refers to enemies with unique health bars that do not drop
  memories.

* NPC quest items are always in the first location you can get them _without_
  killing the NPC or ending the quest early.

### Tutorial

The area from spawning into the starting well up to losing your arm in the intro
boss cutscene.

* **Moon-View Tower:** The pagoda Kuro resides in at the start of the game,
  where he gives you the Kusabimaru.

### Dilapidated Temple

The area where you wake up after the intro boss cutscene. Houses the Sculptor
and leads to Ashina Outskirts.

### Ashina Outskirts

This area ends at the miniboss guarding the last gate to Ashina Castle.
"Left" and "right" are used with the player facing in the direction of area 
progression.

* **Gate:** Large torii gates with solid doors.

* **Wide Gate House:** Large gate with a wide second floor, just after the
  Outskirts Wall - Gate Path idol.

* **Hidden Wall Cave:** A small cave reached by dropping down onto a ledge

>   from the platform left of the Wide Gate House.

Reviewer: I'm not sure this needs its own name. I'd probably just say "cave beneath wide gate house"

Author: works for me, removed and adjusted the Location.

* **Cliff Courtyard:** The miniboss arena after the Outskirts Wall - Gate Path
  idol.

* **Destroyed House:** The two-floor building with a missing roof after the
  cliff courtyard.

* **Cliffside Gate:** The closed gate flanked by giant cobblestone structures,
  next to the secluded platforms with the giant snake skin.

* **Cobblestone Platform:** Large rectangular cobblestone platforms beside a
  giant gate near the Outskirts Wall - Stairway idol.

* **Lookout Building:** The long second floor building that the player grapples
  into after the miniboss blocking progression at the Outskirts Wall - 
  Stairway idol.

* **Headless Cave:** The cliffside cave containing a headless miniboss in
  vanilla.

* **Fortress:** The fortress after the Ashina Castle Gate Fortress idol.

* **Tengu Tower:** The tower accessed after defeating the boss in Gyoubu's
  arena, where you can meet Tengu for the first time.

* **Castle Gate Courtyard:** The courtyard right before the stairs to the Ashina
  Castle Gate miniboss arena.

> * **Ashina Castle entrance:** The miniboss arena right before reaching the

Reviewer: ```suggestion
* **Ashina Castle Entrance:** The miniboss arena right before reaching the
```

Author: Applied and part of next commit.

  shortly before Anayama's final location.

* **Secluded Platforms:** A pair of secluded rock formations to the right just 
  before the cliffside gate. A giant piece of snake skin shed by the Great 
  Serpent hangs off a tree there.

* **Cobblestone Platform:** Large rectangular cobblestone platforms beside a
  giant gate near the Outskirts Wall - Stairway idol.

* **Lookout Building:** The long second floor building that the player grapples
  into after the miniboss blocking progression at the Outskirts Wall - 
  Stairway idol.

* **Headless Cave:** The cliffside cave containing a headless miniboss in
  vanilla.

* **Fortress:** The fortress after the Ashina Castle Gate Fortress idol.

* **Tengu Tower:** The tower accessed after defeating the boss in Gyoubu's
  arena, where you can meet Tengu for the first time.

* **Castle Gate Courtyard:** The courtyard right before the stairs to the Ashina
  Castle Gate miniboss arena.

* **Ashina Castle Entrance:** The miniboss arena right before reaching the
  Ashina Castle idol.

### Hirata Estate (Young Lord's Bell Charm)

* **Estate Path:** The main walled-off path that leads from the Estate Path idol

>   to the rest of the area.

Reviewer: Does this include the courtyards with buildings to either side of the path?

Author: It does now, and I have labeled the courtyard accordingly and added the shortcut to define courtyards after the shortcut, as there are multiple.

* **Side Path Before Cave:** The side path before the cave that leads to Main 
  Hall. Contains two Monk enemies in vanilla.

* **Main Hall:** The area between the Main Hall idol and the miniboss guarding
  the Audience Chamber.

* **Audience Chamber:** The big building containing the Audience Chamber idol
  guarded by a miniboss and many enemies.

* **Hidden Temple:** The boss arena accessed via the Hidden Temple Key at the

>   end of the area.

Reviewer: Do we need a separate area name for this or can we just say "boss arena"?

Author: Could just be boss arena, I just called it Hidden Temple as it is actually called that in the game too. Either way works for me, so I will just remove Hidden Temple and make it boss arena.

  Castle Gate miniboss arena.

* **Ashina Castle Entrance:** The miniboss arena right before reaching the
  Ashina Castle idol.

### Hirata Estate (Young Lord's Bell Charm)

* **Estate Path:** The path that leads from the Estate Path idol to the bridge 
  before the first miniboss. Includes all areas directly connected to the path 
  walls. "Left" and "right" are used with the player facing in the direction of 
  area progression.

    * Courtyard: Walled off areas reachable from the actual path.
    * Shortcut: Areas after the shortcut doors.

* **Bonfire:** The large campfire in a courtyard right of the Estate Path. 
  Surrounded by multiple enemies.

* **Three-Story Pagoda:** The pagoda containing the Mist Raven's Feathers in 
  vanilla.

* **Side Path Before Cave:** The side path before the cave that leads to Main 
  Hall. Contains two Monk enemies in vanilla.

* **Main Hall:** The area between the Main Hall idol and the miniboss guarding
  the Audience Chamber.

* **Audience Chamber:** The big building containing the Audience Chamber idol
  guarded by a miniboss and many enemies.

* **Underwater:** Requires diving via Mibu Breathing Technique.

### Ashina Castle

> This region begins going out the gate after the Ashina Castle Gate idol. "Left" 

Reviewer: Because the boundaries of this region, especially with Ashina Reservoir and Sunken Valley, are pretty ambiguous I think it would be a good idea to be a bit more explicit about where the boundaries are. Maybe something like "It covers everything within the walls of the castle town, as well as the Old Grave area." You'll also need to find a clear line to draw between where "old grave" ends and "sunken valley" begins.

Author: Will update this. Sunken valley begins once you drop into the ravine beyond the Serpent Shrine idol, so I will also add that accordingly.

and "right" are always considering the player to be facing towards Ashina 
Castle.

* **Main Stairway:** The long stairway leading up to the front entrance of the
  castle with a miniboss at the top.

* **Upper Tower:** Anything in the area of the idols that have "Upper Tower" in

>   their naming. Mainly used for the area before Ashina Dojo starting from the 

Reviewer: ```suggestion
  their names. Mainly used for the area before Ashina Dojo starting from the 
```

I would also get rid of the "Mainly used" sentence, if it's not universally true I don't think it's very helpful.

Author: Adjusted location description.

>   Antechamber idol.

Reviewer: This description is kind of confusing in a few ways. "Near these idols" doesn't give very firm boundaries, so it's ambiguous whether this refers to only items indoors or also items on the outside of that tower, which in turn raises the question of whether it refers to _all_ the rooftops of the castle proper or just those of the central tower itself. I think both the definitions and the location descriptions could use some clarification of these questions.

Author: Adjusted location description and definitions.

>     * Roof: The boss arena at the top of the tower.

Reviewer: "Roof" is a kind of confusing term for a place that also has its own roof, and so isn't actually the true roof of the entire castle. As with other regions, I'd just say "boss arena", since again there's only one full memory boss in this region.

Author: Makes sense, will change. I do not think this needs an explicit location description then as that just states the same thing?

Reviewer: I agree.
